axon-server-connector/src/main/java/org/axonframework/axonserver/connector/command/AxonServerCommandBusConnector.java

```diff
@@ -0,0 +1,297 @@
+                        commandPayload.getData().toByteArray(),
+                        convertFromGrpcMetaDataValues(command.getMetaDataMap())
+                ),
+                routingKey,
+                priority
+        );
+    }
+
+    private CommandResponse createResult(Command command, Message<?> resultMessage) {
+        if (resultMessage == null) {
+            return CommandResponse.newBuilder()
+                                  .setMessageIdentifier(UUID.randomUUID().toString())
+                                  .build();
+        }
+
+        String messageId = getOrDefault(resultMessage.getIdentifier(), UUID.randomUUID().toString());
+        CommandResponse.Builder responseBuilder = CommandResponse
+                .newBuilder()
+                .setMessageIdentifier(messageId)
+                .putAllMetaData(AxonServerMetadataConverter.convertToGrpcMetaDataValues(resultMessage.getMetaData()))
+                .setRequestIdentifier(command.getMessageIdentifier());
+        if (!(resultMessage.getPayload() instanceof byte[] payloadAsBytes)) {
+            throw new IllegalArgumentException(
+                    "Payload must be of type byte[] for AxonServerConnector, but was: %s".formatted(
+                            resultMessage.getPayload().getClass().getName())
+            );
+        }
+        return responseBuilder.setPayload(SerializedObject.newBuilder()
+                                                       .setType(resultMessage.type().name())
+                                                       .setRevision(resultMessage.type().version())
+                                                          .setData(ByteString.copyFrom(payloadAsBytes)))
+                              .build();
+    }
+
+    @Override
+    public boolean unsubscribe(@Nonnull QualifiedName commandName) {
+        Registration subscription = subscriptions.remove(commandName);
+        if (subscription != null) {
+            subscription.cancel();
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void onIncomingCommand(@Nonnull CommandBusConnector.Handler handler) {
+        this.incomingHandler.set(handler);
+    }
+
+    private class FutureResultCallback implements ResultCallback {
+
+        private final CompletableFuture<CommandResponse> result;
+        private final Command command;
+
+        public FutureResultCallback(CompletableFuture<CommandResponse> result, Command command) {
+            this.result = result;
+            this.command = command;
+        }
+
+        @Override
+        public void onSuccess(Message<?> resultMessage) {
+            logger.info("Command [{}] completed successfully with result [{}]", command.getName(), resultMessage);
+            result.complete(createResult(command, resultMessage));
+        }
+
+        @Override
+        public void onError(@Nonnull Throwable cause) {
+
```

Comment:
A log statement here might also be nice. Given that we have it on the `onSuccess` method here.